### ISSUE TITLE
SVG background icons not shown in Windows High Contrast Mode

### ISSUE THREAD

**Issue by patrickhlauke** [twbs/bootstrap#21269]
successor to https://github.com/twbs/bootstrap/pull/21268 for v4. while the way icons (such as the hamburger menu) are coded is different (SVG background rather than 3 empty spans), these backgrounds (likely in all other places where SVG backgrounds are currently used) still disappear in WHCM.

<img width="960" alt="capture" src="https://cloud.githubusercontent.com/assets/895831/20806645/219a421a-b7fc-11e6-9d92-75f8e48d600d.PNG">

**Comment by joe-watkins** [comment]
Tnx for firing this up @patrickhlauke :) 

Are foreground `.svg` images an option for critical controls like this?

**Comment by patrickhlauke** [comment]
they'll probably have to be. or we provide some high contrast media query stuff https://msdn.microsoft.com/en-us/library/windows/apps/hh465764.aspx or similar...

**Comment by joe-watkins** [comment]
CSS generated content with the media query could be an option if foreground images aren't possible. May be too opinionated for a generic control.. maybe a warning comment in the docs would do?

``` CSS
@media screen and (-ms-high-contrast: active) {
    .navbar-toggler {
        width: auto;
    }
    .navbar-toggler:after {
        content: "menu";
    }
}
```

![mq-hack](https://cloud.githubusercontent.com/assets/3695795/20846863/783725dc-b880-11e6-8bb9-60f5b5d7d4af.png)

**Comment by patrickhlauke** [comment]
Edging towards a possible solution. First problem is that SVG colors are not automagically inverted when in WHCM. This then results in semi-transparent black hamburger icon on a black background.

http://codepen.io/patrickhlauke/pen/KNoNwd

**Comment by patrickhlauke** [comment]
With some noodling, I tried moving the stroke-related styles to an actual `<style>` block in the SVG, and adding an explicit `-ms-high-contrast` media feature query. The result is semi-promising (though IE refuses to show any background images in WHCM, so a different approach, like CSS generated image content, would have to be used) http://codepen.io/patrickhlauke/pen/WozRyM

**Comment by patrickhlauke** [comment]
![capture](https://cloud.githubusercontent.com/assets/895831/20890076/881b7708-bafd-11e6-9610-f89db8774126.PNG)

**Comment by patrickhlauke** [comment]
more progress using a subtly different approach...but this fails annoyingly in Firefox http://codepen.io/patrickhlauke/pen/bBMbXp

**Comment by fuzzy76** [comment]
#24205 is related I guess. My vote goes to foreground SVG.

**Comment by mdo** [comment]
Is this still an issue @patrickhlauke? Some quick testing for me shows all the navbar toggler icons on the Navbars example pages at https://getbootstrap.com/docs/4.5/examples/navbars/ and https://getbootstrap.com/docs/5.0/examples/navbars/.

**Comment by patrickhlauke** [comment]
I believe this can be closed - at least for the hamburger menu icon, not tested every other icon we have. might open more specific/targeted fresh issues if we stumble across them.

![IE11 showing bootstrap 4.5 documentation, Chrome-based Edge and Firefox showing getbootstrap and bootstrap 4.5 documentation - in WHCM, the hamburger is visible](https://user-images.githubusercontent.com/895831/104121983-3063fd80-533a-11eb-94e5-99cd289d107b.png)
